xarray/core/formatting.py

```diff
@@ -7,6 +7,13 @@
 from .options import OPTIONS
 from .pycompat import iteritems, unicode_type, bytes_type, dask_array_type
 
+def _str_or_unicode(obj):
+    """A drop in replacement for str() which properly handles unicode.
+    """
+    if isinstance(obj, unicode_type):
```

Comment:
I think we actually want to handle this the other way around: convert everything to unicode (str on Python 3). Maybe something like:
```
if isinstance(obj, bytes):
    obj = obj.decode('ascii')
return unicode_type(obj)
```